sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala

```diff
@@ -501,27 +501,22 @@ class MicroBatchExecution(
     // Rewire the plan to use the new attributes that were returned by the source.
     val newAttributePlan = newBatchesPlan transformAllExpressions {
       case ct: CurrentTimestamp =>
+        // CurrentTimestamp is not TimeZoneAwareExpression while CurrentBatchTimestamp is.
+        // Without TimeZoneId, CurrentBatchTimestamp is unresolved. Here, we use an explicit
+        // dummy string to prevent UnresolvedException and to prevent to be used in the future.
         CurrentBatchTimestamp(offsetSeqMetadata.batchTimestampMs,
-          ct.dataType)
+          ct.dataType, Some("Dummy TimeZoneId"))
```

Comment:
We may try to use `DateTimeUtils.defaultTimeZone().getID()` or `conf.sessionLocalTimeZone` (as `ResolveTimeZone` does for `TimeZoneAwareExpression`).

However, `IncrementalExecution` doesn't use timezone info in case of `CurrentBatchTimestamp(_, TimestampType, _)`. And, we want to prevent to use `TimeZone` for this `CurrentTimestamp` expression because this is originally non-`TimeZoneAwareExpression`.

Reply:
Thanks for providing nice analysis and looks like this patch is simpler and more concise. Nice!